Add notification deprecation callout (#2368)
DataDog/terraform-provider-datadog: docs/resources/cloud_configuration_rule.md

### Optional

- `filter` (Block List) Additional queries to filter matched events before they are processed. Defaults to empty list (see [below for nested schema](#nestedblock--filter))

> - `group_by` (List of String) Fields to group by when generating signals, e.g. @resource. Defaults to empty list. Soon to be deprecated in favor of notification rules.

Reviewer: ```suggestion
- `group_by` (List of String) Fields to group by when generating signals, e.g. @resource. This function will be deprecated soon. Use the notification rules function instead.
```

> - `notifications` (List of String) Notification targets for signals. Defaults to empty list. Soon to be deprecated in favor of notification rules.

Reviewer: ```suggestion
- `notifications` (List of String) Notification targets for signals. Defaults to empty list. This function will be deprecated soon. Use the notification rules function instead.
```
